### PR TITLE
KOGITO-4644 Support Quarkus fast-jar

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-4644

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by radtriste** [comment]
testing in progress

File: docs/GUIDE_TO_KOGITO_DEVS.md [hunk 1]
```diff
@@ -103,8 +107,12 @@ Now create a Dockerfile inside the `kogito-examples/process-business-rules-quark
 FROM quay.io/kiegroup/kogito-runtime-jvm:latest
 ENV RUNTIME_TYPE quarkus
 
-COPY target/*-runner.jar $KOGITO_HOME/bin    
-COPY target/lib $KOGITO_HOME/bin/lib
+ENV RUNTIME_TYPE quarkus
```
Comment (sutaakar) [inline docs/GUIDE_TO_KOGITO_DEVS.md:110]: A duplication of line 104/108

File: examples/quarkus-jvm.Dockerfile [hunk 2]
```diff
@@ -10,5 +10,13 @@ ENV RUNTIME_TYPE quarkus
 # 5. Push it: docker push quay.io/<yournamespace>/process-quarkus-example:latest
 # 6. Deploy it on Kubernetes with the kogito operator, as a reference use process-quarkus-example-runtime.yaml
 
-COPY target/*-runner.jar $KOGITO_HOME/bin
-COPY target/lib $KOGITO_HOME/bin/lib
\ No newline at end of file
+ENV RUNTIME_TYPE quarkus
```
Comment (sutaakar) [inline examples/quarkus-jvm.Dockerfile:13]: Same here

File: test/framework/dockerfile_provider.go [hunk 3]
```diff
@@ -16,16 +16,23 @@ package framework
...
 	"strings"
+
+	"github.com/kiegroup/kogito-cloud-operator/core/infrastructure"
 )
 
 const (
-	quarkusJVMApplicationBinarySuffix    = "-runner.jar"
-	quarkusNativeApplicationBinarySuffix = "-runner"
-	springBootApplicationBinarySuffix    = ".jar"
+	protobufFolder = "target/META-INF/resources/persistence/protobuf/"
```
Comment (sutaakar) [inline test/framework/dockerfile_provider.go:27]: Seems unused.

**Comment by radtriste** [comment]
Solving the problem with binary build ...

**Comment by radtriste** [comment]
all binary build/source files passed on my machine for non native. launching the big testing to be sure.

File: cmd/kogito/command/converter/runtimetype_converter.go [hunk 4]
```diff
@@ -40,12 +46,34 @@ func FromArgsToRuntimeType(flags *flag.RuntimeTypeFlags, resourceType flag.Resou
...
 			}
 		}
 		return api.SpringBootRuntimeType, nil
 	}
 
 	return runtimeType, nil
 }
+
+// ToQuarkusLegacyJarType determines what the type of quarkus jar is based on resources
+func ToQuarkusLegacyJarType(resourceType flag.ResourceType, resource string) (bool, error) {
```
Comment (radtriste) [inline cmd/kogito/command/converter/runtimetype_converter.go:62]: I put here for now. Could be moved later.

File: cmd/kogito/command/flag/binary_build_type.go [hunk 5]
```diff
@@ -22,8 +22,10 @@ const (
 	SourceToImageBuild BinaryBuildType = "SourceToImageBuild"
 	// BinarySpringBootJvmBuild is a Spring Boot JVM binary build
 	BinarySpringBootJvmBuild BinaryBuildType = "BinarySpringBootJvmBuild"
-	// BinaryQuarkusJvmBuild is a Quarkus JVM binary build
-	BinaryQuarkusJvmBuild BinaryBuildType = "BinaryQuarkusJvmBuild"
+	// BinaryQuarkusFastJvmBuild is a Quarkus JVM binary build with Fast Jar
+	BinaryQuarkusFastJvmBuild BinaryBuildType = "BinaryQuarkusFastJvmBuild"
```
Comment (spolti) [inline cmd/kogito/command/flag/binary_build_type.go:26]: maybe add FastJar and LEgacyJar on the names for a better understanding?

File: cmd/kogito/command/converter/runtimetype_converter.go [hunk 6]
```diff
@@ -15,10 +15,16 @@
...
 	"io/ioutil"
 	"strings"
+
+	"github.com/kiegroup/kogito-cloud-operator/api"
+	"github.com/kiegroup/kogito-cloud-operator/cmd/kogito/command/flag"
+)
+
+var (
+	quarkusLegacyJarSuffix = "-runner.jar"
+	quarkusRuntimeSuffixes = []string{quarkusLegacyJarSuffix, "quarkus-app", "-runner"}
```
Comment (spolti) [inline cmd/kogito/command/converter/runtimetype_converter.go:27]: shouldn't it be quarkus-run or this is to identify the dir quarkus-app for fast Jar?

**Review by spolti — CHANGES_REQUESTED** [review]

**Comment by spolti** [comment]
@radtriste also, I think you forget to add the jira on the release notes.

**Review by ricardozanini — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]
Looks good to me.

File: cmd/kogito/command/converter/runtimetype_converter.go (hunk 6 above)
Comment (radtriste) [reply cmd/kogito/command/converter/runtimetype_converter.go:27]: renamed to `quarkusRuntimeTargetSuffixes`. Indeed this is to check in the `target` folder

**Comment by radtriste** [comment]
Looks like latest run of nightly happened already and broke something ... opened https://github.com/kiegroup/kogito-pipelines/pull/130 to mitigate that problem and relaunch the nightly pipeline to check .,,

**Comment by radtriste** [comment]
Apart from that full testing was ok on this specific branch => https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/custom/job/tradisso/job/dsl/job/nightly/job/kogito-4644/job/kogito-operator-deploy/2/
Gonna review the unit tests and covergae errors

**Review by spolti — APPROVED** [review]

**Comment by radtriste** [comment]
codecov finished with a weird authentication error: `{'detail': ErrorDetail(string='Could not find a repository, try using upload token', code='not_found')}`
Merging this one as full testing was ok and all approved so it can be in the next nightly run

**Comment by ricardozanini** [comment]
@radtriste just a small remark: if you do `kogito deploy my-service target/quarkus-app` won't work. If we have documentation about this procedure, we should emphasize that. Many thanks for this! Really helped!

**Comment by radtriste** [comment]
Our documentation is already emphasizing to upload directly from `target/` folder. But I guess the docs need to be updated with the `fast-jar` jar files naming and precise it. gonna PR `kie-docs` for that.
Also just realized I forgot to update the release notes => https://github.com/kiegroup/kogito-operator/pull/793

**Comment by radtriste** [comment]
docs update => https://github.com/kiegroup/kie-docs/pull/3363

**Pull request merged**